### DISCUSSION TITLE
Is there any way to get the notification about reaching the rate limits?

### DISCUSSION THREAD

**Discussion by siarheipashkevich** [laravel/framework#50898] [Q&A] [answered]
Hi, all!

Do you know how to get notification if the any rate limits is reached?
https://laravel.com/docs/11.x/routing#rate-limiting

Thanks

**Comment by ghost** [comment, marked as answer]
```
RateLimiter::for('global', function (Request $request) {
    return Limit::perMinute(1000)->response(function (Request $request, array $headers) {
        // do your magic here.
        return response('Custom response...', 429, $headers);
    });
});
```

**Reply by siarheipashkevich** [reply]
I saw this solution, but I want to save the initial response of the Laravel and don't rewrite it

**Reply by ghost** [reply]
Do a test to see that response and put the same text there. I would leave the text empty string and leave just the 429.

**Reply by siarheipashkevich** [reply]
I have global handler for exceptions and there is one which catch `ThrottleRequestsException` for some preparation. I'm not sure that this type of the exception will be the same after transforming rate limit response as you suggested above.

**Reply by siarheipashkevich** [reply]
<img width="976" alt="image" src="https://github.com/laravel/framework/assets/7753600/d1a2b8e8-fab0-49c3-8cd2-4ce1145a1d48">

**Reply by siarheipashkevich** [reply]
<img width="1079" alt="image" src="https://github.com/laravel/framework/assets/7753600/82d0fbe5-6497-4553-8a78-fd1664e24491">

**Reply by ghost** [reply]
Limit::perMinute(1000)->response( ...  should be Too Many Attempts.

**Reply by siarheipashkevich** [reply]
I described the reason why I can't do it

**Reply by ghost** [reply]
As I see it, if you use the callback to transform the response, then it will return HttpResponseException. If not it will return ThrottleResponseException. You are concerned that the first case will contain other message. The rate limiting should be executed first before the controller so what exception can there be? And if there is one, it will not be triggered by your rate limit callback. It will have its own callback or none.

**Reply by siarheipashkevich** [reply]
Thanks for your response, I understood how to implement it using global exception handler (but there is no information about name of the rate limit. `ThrottleRequestsException` does not contain for which rate limit it was thrown).

**Comment by siarheipashkevich** [comment]
I'm talking about something like:

https://laravel.com/docs/11.x/database#monitoring-cumulative-query-time

or `Model::handleLazyLoadingViolationUsing(...)`

But only for reaching Rate Limits.
